.github/workflows/test-pg_search.yml

```diff
@@ -90,18 +90,21 @@ jobs:
           fi
```

Comment:
Could you do the same you did here for the caching in the `lint-rust.yml` workflow? It's also hella slow

Reply:
what are you actually talking about?

Reply:
Our Lint Rust workflow also needs to have the way it does caching updated

Reply:
I can take a look at it once I feel like I've got this file doing what I hope it can do.